v2/internal/reconcilers/generic/generic_reconciler.go

```diff
@@ -3,7 +3,7 @@ Copyright (c) Microsoft Corporation.
 Licensed under the MIT license.
 */
 
-package controllers
+package generic
```

Comment:
I thought about it more and I think I like the `internal/reconcilers/generic` path, over `internal/controllers`.

That does mean:
`internal/reconcilers/generic` runs inner reconcilers of types `internal/reconcilers/arm` and `internal/reconcilers/mysql` (and in the future more), BUT I don't think that's all that confusing and it puts all the reconciliation stuff next to one another which makes a lot of sense.